lib/puppet/resource/catalog.rb

```diff
@@ -536,6 +546,7 @@ def to_catalog(convert)
 
     result.version = self.version
     result.code_id = self.code_id
+    result.catalog_uuid = self.catalog_uuid
```

Comment:
Can we add a spec test that the copy constructor behavior preserves the catalog_uuid?